drivers/dac/dac_demo/dac_demo.c

```diff
@@ -85,6 +92,128 @@ int32_t dac_demo_remove(struct dac_demo_desc *desc)
 	return SUCCESS;
 }
 
+
+int32_t dac_get_number_of_channels(void* dev)
+{
+	struct dac_demo_desc *desc = dev;
+
+	return desc->active_ch;
+}
+
+/***************************************************************************//**
+ * @brief update number of active channels
+ * @param dev - physical instance of a dac device
+ * @param mask - the new number of active channels
+ * @return SUCCESS in case of success.
+*******************************************************************************/
+int32_t update_active_dac_channels(void *dev, int32_t mask)
+{
+	struct dac_demo_desc *desc = dev;
+	desc->active_ch = mask;
+
+	return SUCCESS;
+}
+
+/***************************************************************************//**
+ * @brief close all channels
+ * @param dev - physical instance of an adc device
+ * @return SUCCESS in case of success.
+*******************************************************************************/
+int32_t close_dac_channels(void* dev)
+{
+	struct dac_demo_desc *desc = dev;
+	desc->active_ch = 0;
+
+	return SUCCESS;
+}
+
+/***************************************************************************//**
+ * @brief utility function for computing next upcoming channel
+ * @param ch_mask - active channels .
+ * @param last_idx -  previous index.
+ * @param new_idx - upcoming channel index, return param.
+ * @return 1 if there are more channels, 0 if done.
+*******************************************************************************/
+static bool get_next_ch_idx(uint32_t ch_mask, uint32_t last_idx,
+			    uint32_t *new_idx)
+{
+	last_idx++;
+	ch_mask >>= last_idx;
+	if (!ch_mask) {
+		*new_idx = -1;
+		return 0;
+	}
+	while (!(ch_mask & 1)) {
+		last_idx++;
+		ch_mask >>= 1;
+	}
+	*new_idx = last_idx;
+
+	return 1;
+}
+
+int32_t dac_write_samples(void* dev, uint16_t* buff, uint32_t samples)
```

Comment:
Add function header comment.

Also, maybe return an error code instead or `samples`.

Otherwise, you should switch the return type for the function to `uint32_t` since the number of samples cannot be negative.

Reply:
The function should return the number of samples because they are used, somewhere else, but I changed it to uin32_t